cheddar/cheddar-integration-aws/src/main/java/com/clicktravel/infrastructure/persistence/aws/dynamodb/DynamoDbTemplate.java

```diff
@@ -642,4 +621,64 @@ public void delete(final Item item, final PersistenceExceptionHandler<?>... pers
         }
     }
+
+    private boolean isQueryOnIndex(final AttributeQuery query, final ItemConfiguration itemConfiguration,
+            final String queryAttributeName) {
+        return !(itemConfiguration.primaryKeyDefinition().propertyName().equals(queryAttributeName)
+                && !(query instanceof CompoundAttributeQuery));
+    }
+
+    private <T extends Item> Map<String, com.amazonaws.services.dynamodbv2.model.Condition> createDynamoDbConditionsMap(
+            final AttributeQuery query, final ItemConfiguration itemConfiguration) throws Exception {
+        final Map<String, com.amazonaws.services.dynamodbv2.model.Condition> conditions = new HashMap<>();
+
+        final com.amazonaws.services.dynamodbv2.model.Condition condition = createDynamoDbCondition(
+                query.getCondition(), query.getAttributeName(), itemConfiguration);
+        conditions.put(query.getAttributeName(), condition);
+
+        if (CompoundAttributeQuery.class.isAssignableFrom(query.getClass())) {
+            final CompoundAttributeQuery compoundAttributeQuery = (CompoundAttributeQuery) query;
+            final com.amazonaws.services.dynamodbv2.model.Condition supportingCondition = createDynamoDbCondition(
+                    compoundAttributeQuery.getSupportingCondition(),
+                    compoundAttributeQuery.getSupportingAttributeName(), itemConfiguration);
+            conditions.put(compoundAttributeQuery.getSupportingAttributeName(), supportingCondition);
+        }
+
+        return conditions;
+    }
+
+    private <T extends Item> com.amazonaws.services.dynamodbv2.model.Condition createDynamoDbCondition(
+            final Condition condition, final String propertyName, final ItemConfiguration itemConfiguration)
+                    throws Exception {
+        final com.amazonaws.services.dynamodbv2.model.Condition dynamoDbCondition = new com.amazonaws.services.dynamodbv2.model.Condition();
+
+        if (condition.getComparisonOperator() == Operators.NULL) {
+            dynamoDbCondition.setComparisonOperator(ComparisonOperator.NULL);
+        } else if (condition.getComparisonOperator() == Operators.NOT_NULL) {
+            dynamoDbCondition.setComparisonOperator(ComparisonOperator.NOT_NULL);
+        } else {
+            if (condition.getComparisonOperator() == Operators.EQUALS) {
+                dynamoDbCondition.setComparisonOperator(ComparisonOperator.EQ);
+            } else if (condition.getComparisonOperator() == Operators.LESS_THAN_OR_EQUALS) {
+                dynamoDbCondition.setComparisonOperator(ComparisonOperator.LE);
+            } else if (condition.getComparisonOperator() == Operators.GREATER_THAN_OR_EQUALS) {
+                dynamoDbCondition.setComparisonOperator(ComparisonOperator.GE);
+            }
+
+            final Collection<AttributeValue> attributeValueList = new ArrayList<>();
+
+            for (final String stringValue : condition.getValues()) {
```

Comment:
I think this in this instance, it will actually meant the code harder to read and understand.  Unless you have a suggestion for something which won't?